### PR TITLE
Update GetDataHandlers.cs

### LINKED ISSUE #2804: Terra Blade triggered MaxProjDamage Etc
TShock version: TShock 5

This is probably caused by GetDataHandlers.cs>HandleProjectileNew not updating.

In addition, many of GetDataHandlers.cs have not been updated, such as GetDataHandlers.cs>HandleSpawn

### PR DESCRIPTION
**Pull request by ATFGK**
Fixes #2804 

Fix GetDataHandlers.cs file for version 144X, and correct various errors caused by it.

### CODE REVIEW

**Comment by drunderscore** [comment]
As originally requested in #2823, please don't include changes to number of PVE/PVP deaths, as that will be dealt with in #2768

Commits and changes are still a bit scattered, and it's not super clear which issues this PR solves -- can you update the description to include the specific issues that this PR solves?

**Comment by ATFGK** [comment]
> As originally requested in #2823, please don't include changes to number of PVE/PVP deaths, as that will be dealt with in #2768
> 
> Commits and changes are still a bit scattered, and it's not super clear which issues this PR solves -- can you update the description to include the specific issues that this PR solves?

1.Fixed wrong MaxProjDamage trigger.
2.Fix the wrong resurrection prompt.
Once I delete the content related to PVE/PVP, the second problem will not be solved.

**Comment by drunderscore** [comment]
@ATFGK I'm not sure I understand that last issue -- can you describe it further or show an example?

**Comment by ATFGK** [comment]
> @ATFGK I'm not sure I understand that last issue -- can you describe it further or show an example?

It has little impact on "Tshock", but the wrong "SpawnContext" message will cause a lot of problems for other plug-ins because they cannot correctly recognize "SpawnContext"

**Comment by ATFGK** [comment]
> > @ATFGK I'm not sure I understand that last issue -- can you describe it further or show an example?
> 
> It has little impact on "Tshock", but the wrong "SpawnContext" message will cause a lot of problems for other plug-ins because they cannot correctly recognize "SpawnContext"

"Tshock" also uses a lot of "TSPlayer. Spawn()", so other plug-ins cannot avoid receiving incorrect "SpawnContext"

**Comment by drunderscore** [comment]
Yeah, totally see the issue and understand. With that other PR blocked for non-working, I'm fine accepting these changes first, even if it means the other PR will get conflicts.

I'll review this ASAP.

File: docs/changelog.md [hunk 1]
```diff
@@ -81,15 +81,15 @@ Use past tense when adding new entries; sign your name off when you add or chang
...
-* Update argument `npcIndex` in `GetDataHandlers.HandleNpcTeleportPortal` to `ushort`.(@ATFGK)
+* New argument `AI[2]` has been added to hook `GetDataHandlers.HandleProjectileNew`.(@ATFGK)
+* New argument `Zone5` has been added to hook `GetDataHandlers.OnPlayerZone`.(@ATFGK)
+* New argument `coatTile` has been added to hook `GetDataHandlers.OnPaintTile`.(@ATFGK)
+* New argument `coatWall` has been added to hook `GetDataHandlers.OnPaintWall`.(@ATFGK)
+* New argument `CooldownCounter` has been added to hook `GetDataHandlers.OnPlayerDamage`.(@ATFGK)
+* New argument `Random` has been added to hook `GetDataHandlers.OnPlaceObject`.(@ATFGK)
+* New argument `numberOfDeathsPVE` and `numberOfDeathsPVP` have been added to hook `GetDataHandlers.OnPlayerSpawn`.(@ATFGK)
+* New argument `numberOfDeathsPVE` and `numberOfDeathsPVP` have been added to `TSPlayer.Spawn`.(@ATFGK)
+* An argument `npcIndex` in `GetDataHandlers.HandleNpcTeleportPortal` has been updated to `ushort`.(@ATFGK)
```
Comment (drunderscore) [inline docs/changelog.md]: These should all be placed into the past tense, starting with the verb instead:
> Added new argument(s) (description of change)...

File: TShockAPI/Net/SpawnMsg.cs [hunk 2]
```diff
@@ -33,6 +33,8 @@ public override PacketTypes ID
 		public short TileX { get; set; }
 		public short TileY { get; set; }
 		public int RespawnTimer { get; set; }
+		public short numberOfDeathsPVE { get; set; }
+		public short numberOfDeathsPVP { get; set; }
```
Comment (drunderscore) [inline TShockAPI/Net/SpawnMsg.cs]: Use `UpperCamelCase` instead

**Review by drunderscore — CHANGES_REQUESTED** [review]
All the code looks good -- just some code smell and grammatical issues otherwise!

**Review by drunderscore — CHANGES_REQUESTED** [review]
Unfortunately the changelog quality is still quite lacking -- so instead of holding this PR up for any longer due to it's non-conforming changelog, please copy the changelog entries from [this changelog file](https://raw.githubusercontent.com/drunderscore/TShock/atfgk-proper-changelog/docs/changelog.md) into yours.

**Comment by drunderscore** [comment]
To make it even easier, I've created a [pull request on your repository](https://github.com/ATFGK/TShock/pull/33) for this branch with the correct changes.

**Comment by ATFGK** [comment]
> To make it even easier, I've created a [pull request on your repository](https://github.com/ATFGK/TShock/pull/33) for this branch with the correct changes.

Thank you very much for your help. :-D

**Pull request merged**